### PR TITLE
Add missing license headers to benchmarks and tests

### PR DESCRIPTION
**Pull request by jacobhinkle**
Just noticed some files were missing license notices.

This PR adds those missing licenses and also creates a new CI job that greps for the copyright notice on new `*.{cpp,h,py}` files.

### CODE REVIEW

**Comment by jacobhinkle** [comment]
This command helps find missing headers.
```shell
find /opt/pytorch/nvfuser -regextype posix-extended -regex '.*\.(cpp|h|py)' \
  | egrep -v '^/opt/pytorch/nvfuser/(\.|third_party|bin|build|nvfuser/include|nvfuser/version.py|tools/linter/adapters|csrc/serde/fusion_cache_generated.h)' \
  | xargs grep -L SPDX-FileCopyrightTex
```
It currently returns no lines. I have added it as an external CI task.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -14,6 +14,19 @@ env:
...
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Look for missing license headers
+        working-directory: ${{ env.working_directory }}
+        run: |
+          find . -regextype posix-extended -regex '.*\.(cpp|h|py)' | egrep -v '^./(\.|third_party|bin|build|nvfuser/include|nvfuser/version.py|tools/linter/adapters|csrc/serde/fusion_cache_generated.h)' | xargs grep -L SPDX-FileCopyrightText | tee missing-header-files.txt
+          # test that file is empty
+          test ! -s missing-header-files.txt
```
Comment (jjsjann123) [inline .github/workflows/lint.yml:28]: :clap:

**Review by jjsjann123 — APPROVED** [review]
FYI, we have an exclude dir in :https://github.com/NVIDIA/Fuser/blob/main/tools/update_copyright.py

I have to say it's been a while since I used that so not sure if it's still relevant.

**Review by xwang233 — APPROVED** [review]

**Comment by jacobhinkle** [comment]
> FYI, we have an exclude dir in :https://github.com/NVIDIA/Fuser/blob/main/tools/update_copyright.py

Oh wow I somehow missed that that tool even exists. I'll go ahead and merge this just to have something in CI, but we can look at updating that script and maybe using it in the CI job instead, just like how `lintrunner -a` could also be used to detect whether formatting changes are needed.

**Pull request merged**